openmc/deplete/batchwise.py

```diff
@@ -0,0 +1,1157 @@
+        translation or rotation attributes
+    axis : int {0,1,2}
+        Directional axis for geometrical parametrization, where 0, 1 and 2 stand
+        for 'x', 'y' and 'z', respectively.
+    """
+    def __init__(self, cell, operator, model, bracket, bracket_limit, axis=None,
+                 density_treatment='constant-volume', bracketed_method='brentq',
+                 tol=0.01, target=1.0, print_iterations=True,
+                 search_for_keff_output=True):
+
+        super().__init__(operator, model, bracket, bracket_limit, density_treatment,
+                         bracketed_method, tol, target, print_iterations,
+                         search_for_keff_output)
+
+        self.cell = self._get_cell(cell)
+
+        if axis is not None:
+            #index of cell directional axis
+            check_value('axis', axis, (0,1,2))
+        self.axis = axis
+
+        # list of material fill the attribute cell, if depletables
+        self.cell_materials = None
+
+    @abstractmethod
+    def _get_cell_attrib(self):
+        """
+        Get cell attribute coefficient.
+        Returns
+        -------
+        coeff : float
+            cell coefficient
+        """
+
+    @abstractmethod
+    def _set_cell_attrib(self, val):
+        """
+        Set cell attribute to the cell instance.
+        Parameters
+        ----------
+        val : float
+            cell coefficient to set
+        """
+
+    def _get_cell(self, val):
+        """Helper method for getting cell from cell instance or cell name or id.
+        Parameters
+        ----------
+        val : Openmc.Cell or str or int representing Cell
+        Returns
+        -------
+        val : openmc.Cell
+            Openmc Cell
+        """
+        if isinstance(val, Cell):
+            check_value('Cell exists', val, [cell for cell in \
+                                self.geometry.get_all_cells().values()])
+
+        elif isinstance(val, str):
+            if val.isnumeric():
+                check_value('Cell id exists', int(val), [cell.id for cell in \
+                                self.geometry.get_all_cells().values()])
+                val = [cell for cell in \
+                       self.geometry.get_all_cells().values() \
+                       if cell.id == int(val)][0]
+            else:
+                check_value('Cell name exists', val, [cell.name for cell in \
+                                self.geometry.get_all_cells().values()])
+
+                val = [cell for cell in \
+                       self.geometry.get_all_cells().values() \
+                       if cell.name == val][0]
+
+        elif isinstance(val, int):
+            check_value('Cell id exists', val, [cell.id for cell in \
+                                self.geometry.get_all_cells().values()])
+            val = [cell for cell in \
+                       self.geometry.get_all_cells().values() \
+                       if cell.id == val][0]
+        else:
```

Comment:
That's a lot of calls to `self.geometry.get_all_cells` which could be super expensive for large models. Can we instead store all the potential cells at the top of this method?